nbconvert/preprocessors/extractoutput.py

```diff
@@ -73,6 +75,10 @@ def preprocess_cell(self, cell, resources, cell_index):
                 if mime_type in out.data:
                     data = out.data[mime_type]
 
+                    # If data was JSON, it will become a NotebookNode at this point.
+                    if isinstance(data, NotebookNode):
+                        data = json.dumps(data)
```

Comment:
This shouldn't actually be turning it to text, I think. That would do incorrect things for `--to notebook`, for example. I think `ExtractOutputPreprocessor` needs to be taught how to handle when this is a non-string (i.e. json serialize as part of writing to a file), rather than casting it to a string here.

Test ideas:

- convert to notebook and verify that json outputs are preserved as objects
- extract output with json
- make sure to include a variety of json outputs which can be dicts, lists, scalars, etc., not just dicts.

Reply:
> convert to notebook and verify that json outputs are preserved as objects

I believe this was already the case: we're dealing with a preprocessor that doesn't modify the cell outputs.

I have added a test to verify that it is indeed the case.